backend/src/main/java/com/redhat/cloud/notifications/routers/sources/SecretUtils.java

```diff
@@ -0,0 +1,224 @@
+                final Secret secret = this.sourcesService.getById(secretTokenSourcesId);
+
+                props.setSecretToken(secret.password);
+            }
+        }
+    }
+
+    /**
+     * Creates the endpoint's secrets in Sources.
+     * @param endpoint the endpoint to create the secrets from.
+     */
+    public void createSecretsForEndpoint(Endpoint endpoint) {
+        EndpointProperties endpointProperties = endpoint.getProperties();
+
+        if (endpointProperties instanceof SourcesSecretable) {
+            var props = (SourcesSecretable) endpointProperties;
+
+            final BasicAuthentication basicAuth = props.getBasicAuthentication();
+            if (basicAuth != null) {
+                final long id = this.createBasicAuthentication(basicAuth);
+
+                Log.infof("[endpoint_id: %s][secret_id: %s] Basic authentication secret created in Sources", endpoint.getId(), id);
+
+                props.setBasicAuthenticationSourcesId(id);
+            }
+
+            final String secretToken = props.getSecretToken();
+            if (secretToken != null) {
+                final long id = this.createSecretTokenSecret(secretToken);
+
+                Log.infof("[endpoint_id: %s][secret_id: %s] Secret token secret created in Sources", endpoint.getId(), id);
+
+                props.setSecretTokenSourcesId(id);
+            }
+        }
+    }
+
+    /**
+     * <p>Updates the endpoint's secrets in Sources. However a few cases are covered for the secrets:</p>
+     * <ul>
+     *  <li>If the endpoint has an ID for the secret, and the incoming secret is {@code null}, it is assumed that the
+     *  user wants the secret to be deleted.</li>
+     *  <li>If the endpoint has an ID for the secret, and the incoming secret isn't {@code null}, then the secret is
+     *  updated</li>
+     *  <li>If the endpoint doesn't have an ID for the secret, and the incoming secret is {@code null}, it's basically
+     *  a NOP — although the attempt is logged for debugging purposes.</li>
+     *  <li>If the endpoint doesn't have an ID for the secret, and the incoming secret isn't {@code null}, it is
+     *  assumed that the user wants the secret to be created.</li>
+     * </ul>
+     * @param endpoint the endpoint to update the secrets from.
+     */
+    public void updateSecretsForEndpoint(Endpoint endpoint) {
+        EndpointProperties endpointProperties = endpoint.getProperties();
+
+        if (endpointProperties instanceof SourcesSecretable) {
+            var props = (SourcesSecretable) endpointProperties;
+
+            final BasicAuthentication basicAuth = props.getBasicAuthentication();
+            final Long basicAuthId = props.getBasicAuthenticationSourcesId();
+            // If the object is null, we simply assume that the user decided to delete the secret, since an update
+            // operation requires the user to send the full object.
+            //
+            // If it isn't null, then we
```

Comment:
`Then we` what? 😄

Reply:
Oops! :eyes:

Reply:
Actually I've removed the comment since this was already explained in the Javadoc of the method.